sklearn/tree/_criterion.pyx

```diff
@@ -1160,6 +1165,19 @@ cdef class MSE(RegressionCriterion):
                 y_ik = self.y[i, k]
                 sq_sum_left += w * y_ik * y_ik
 
+        # Account for missing values that may be on the left node.
+        # If so, then these samples are still not included in the
+        # sq_sum_left because they are in samples[end_non_missing:end].
+        if self.missing_go_to_left:
```

Comment:
I prefer the comment inside the statement. I don't think we need details regarding the indexing since this explained elsewhere in the paritioning.

```suggestion
        if self.missing_go_to_left:
            # add up the impact of these missing values on the left child
            # statistics.
```